### PR TITLE
PV2-1941 Removed sensitive info from the logs

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

File: src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockStatusChangedEventBatchProcessor.cs [hunk 1]
```diff
@@ -61,10 +61,7 @@ public async Task<int> Process(int batchSize, CancellationToken cancellationToke
                         foreach(var error in changeEvent.Errors)
                             await SaveErrorCode(error, changeEvent, cancellationToken);
 
-                        logger.LogDebug(
-                            $"Saved PriceEpisodeStatusChange event {changeEvent.DataLock.DataLockEventId} for UKPRN {changeEvent.DataLock.UKPRN}. " +
-                            $"Commitment versions: {changeEvent.CommitmentVersions.Length}, " +
-                            $"periods: {changeEvent.Periods.Length}, errors: {changeEvent.Errors.Length}");
+                        logger.LogDebug($"Saved PriceEpisodeStatusChange event {changeEvent.DataLock.DataLockEventId} for UKPRN {changeEvent.DataLock.UKPRN}. Commitment versions: {changeEvent.CommitmentVersions.Length}, periods: {changeEvent.Periods.Length}, errors: {changeEvent.Errors.Length}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.Application/Services/DataLockStatusChangedEventBatchProcessor.cs]: undo whole file

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Handlers/ApprenticeshipCreatedHandler.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ protected override async Task HandleMessage(ApprenticeshipCreatedEvent message,
             Logger.LogDebug($"Handling apprenticeship created event.  Now resolving the apprenticeship processor service to handle the new apprenticeship. Apprenticeship Id: {message.ApprenticeshipId}, Account: {message.AccountId}, Ukprn: {message.ProviderId}");
             var processor = scope.Resolve<IApprenticeshipProcessor>();
             await processor.Process(message);
-            Logger.LogInfo($"Finished handling apprenticeship created event.  Now resolving the apprenticeship processor service to handle the new apprenticeship. Apprenticeship Id: {message.ApprenticeshipId}, Account: {message.AccountId}, Ukprn: {message.ProviderId}");
+            Logger.LogInfo($"Finished handling apprenticeship created event. Now resolving the apprenticeship processor service to handle the new apprenticeship. Apprenticeship Id: {message.ApprenticeshipId}, Account: {message.AccountId}, Ukprn: {message.ProviderId}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/Handlers/ApprenticeshipCreatedHandler.cs:23]: undo whole file

File: src/SFA.DAS.Payments.DataLocks.ApprovalsService/Handlers/ApprenticeshipUpdatedHandler.cs [hunk 3]
```diff
@@ -16,17 +16,13 @@ public ApprenticeshipUpdatedHandler(IPaymentLogger logger, IContainerScopeFactor
 
         protected override async Task HandleMessage(ApprenticeshipUpdatedApprovedEvent message, IMessageHandlerContext context, ILifetimeScope scope)
         {
-            Logger.LogDebug($"Handling apprenticeship updated event.  " +
-                            $"Now resolving the apprenticeship processor service to handle Updated apprenticeship. " +
-                            $"Apprenticeship Id: {message.ApprenticeshipId}");
+            Logger.LogDebug($"Handling apprenticeship updated event. Now resolving the apprenticeship processor service to handle Updated apprenticeship. Apprenticeship Id: {message.ApprenticeshipId}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.ApprovalsService/Handlers/ApprenticeshipUpdatedHandler.cs]: undo whole file

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ApprenticeshipUpdatedHandler.cs [hunk 4]
```diff
@@ -29,7 +29,10 @@ public async Task Handle(ApprenticeshipUpdated message, IMessageHandlerContext c
                 throw new InvalidOperationException("Invalid 'ApprenticeshipUpdated' received. Uln was 0.");
             }
 
-            logger.LogDebug($"Now handling the apprenticeship updated event.  Apprenticeship: {message.Id}, employer: {message.EmployerAccountId}, ukprn: {message.Ukprn}");
+            logger.LogDebug($"Now handling the apprenticeship updated event.  " +
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ApprenticeshipUpdatedHandler.cs]: undo whole file

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ApprenticeshipUpdatedHandler.cs [hunk 5]
```diff
@@ -29,7 +29,7 @@ public async Task Handle(ApprenticeshipUpdated message, IMessageHandlerContext c
                 throw new InvalidOperationException("Invalid 'ApprenticeshipUpdated' received. Uln was 0.");
             }
 
-            logger.LogDebug($"Now handling the apprenticeship updated event.  Apprenticeship: {message.Id}, employer: {message.EmployerAccountId}, ukprn: {message.Ukprn}");
+            logger.LogDebug($"Now handling the apprenticeship updated event. Apprenticeship: {message.Id}, employer: {message.EmployerAccountId}, ukprn: {message.Ukprn}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ApprenticeshipUpdatedHandler.cs:32]: undo whole file

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ResetActorsHandler.cs [hunk 6]
```diff
@@ -28,9 +28,10 @@ public async Task Handle(ResetActorsCommand message, IMessageHandlerContext cont
             foreach (var uln in message.Ulns)
             {
                 var actorId = new ActorId(uln.ToString());
-                logger.LogVerbose($"Creating actor proxy, actor id: {uln}.");
+                // Currently only called from tests. Removing log as it mentions ULN
+                //logger.LogVerbose($"Creating actor proxy, for learner: {message}.");
                 var actor = proxyFactory.CreateActorProxy<IDataLockService>(new Uri("fabric:/SFA.DAS.Payments.DataLocks.ServiceFabric/DataLockServiceActorService"), actorId);
-                logger.LogVerbose($"Actor proxy created. Actor id: {uln}, now resetting the cache.");
+                //logger.LogVerbose($"Actor proxy created. Actor id: {uln}, now resetting the cache.");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ResetActorsHandler.cs:34]: there are two things that can be done here, 

1 remove all LogVerbose like this one from all places in this PR
2 change this LogVerbose to be same as other places in this PR

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ResetCacheHandler.cs [hunk 7]
```diff
@@ -32,7 +32,8 @@ public async Task Handle(ResetCacheCommand message, IMessageHandlerContext conte
             foreach (var uln in ulns)
             {
                 var actorId = new ActorId(uln.ToString());
-                logger.LogVerbose($"Creating actor proxy for actor id: {uln}");
+                // This handler isn't used. Removing log for ULN
+                //logger.LogVerbose($"Creating actor proxy for actor id: {uln}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/ResetCacheHandler.cs:36]: 1 remove all LogVerbose like this one from all places in this PR
2 change this LogVerbose to be same as other places in this PR

File: src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs [hunk 8]
```diff
@@ -123,34 +123,36 @@ public async Task<List<FunctionalSkillDataLockEvent>> GetApprenticeshipUpdateFun
 
         public async Task Reset()
         {
-            paymentLogger.LogVerbose($"Resetting actor for id {Id}");
+            // This method is only used from tests currently. Removing log for ULN
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs:126]: 1 remove all LogVerbose like this one from all places in this PR
2 change this LogVerbose to be same as other places in this PR

File: src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs [hunk 9]
```diff
@@ -123,34 +123,36 @@ public async Task<List<FunctionalSkillDataLockEvent>> GetApprenticeshipUpdateFun
...
-                paymentLogger.LogDebug($"Activating data-lock actor: {Id}");
+                // Not sure what to do about this... it used to log the ID (ULN)
+                paymentLogger.LogDebug($"Activating data-lock actor");
                 var stopwatch = Stopwatch.StartNew();
                 await Initialise().ConfigureAwait(false);
                 await base.OnActivateAsync().ConfigureAwait(false);
                 TrackInfrastructureEvent("DataLockService.OnActivateAsync", stopwatch);
                 telemetry.StopOperation(operation);
-                paymentLogger.LogInfo($"Finished activating data-lock actor: {Id}");
+                paymentLogger.LogInfo($"Finished activating data-lock actor");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs:143]: make it consistent

Comment (carlbm) [reply src/SFA.DAS.Payments.DataLocks.DataLockService/DataLockService.cs:140]: We don't have access to the information at this point. Knowing that we've activated the actor is better than no log at all

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/PeriodEndStartedHandler.cs [hunk 10]
```diff
@@ -21,7 +21,7 @@ public PeriodEndStartedHandler(IPaymentLogger logger)
 
         public async Task Handle(PeriodEndStartedEvent message, IMessageHandlerContext context)
         {
-            logger.LogDebug($"Received period end started event. Details: {message.ToJson()}");
+            logger.LogDebug($"Received period end started event. JobId: {message.JobId}, Collection Period: {message.CollectionPeriod}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/PeriodEndStartedHandler.cs:24]: Please undo this file as discussed

File: src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/PeriodEndStoppedHandler.cs [hunk 11]
```diff
@@ -28,7 +28,7 @@ public PeriodEndStoppedHandler(IPaymentLogger logger, IConfigurationHelper confi
 
         public async Task Handle(PeriodEndStoppedEvent message, IMessageHandlerContext context)
         {
-            logger.LogDebug($"Received period end stopped event. Details: {message.ToJson()}");
+            logger.LogDebug($"Received period end stopped event. JobId: {message.JobId}, Collection Period: {message.CollectionPeriod}");
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.DataLocks.DataLockProxyService/Handlers/PeriodEndStoppedHandler.cs:31]: Please undo this file as discussed

**Pull request merged**